src/gtk/toga_gtk/widgets/detailedlist.py

```diff
@@ -1,92 +1,185 @@
+        self._on_select_signal_handler = None
 
-        self.treeview = Gtk.TreeView(model=self.store)
-        self.treeview.set_headers_visible(False)
-        self.selection = self.treeview.get_selection()
-        self.selection.set_mode(Gtk.SelectionMode.SINGLE)
-        self.selection.connect("changed", self.gtk_on_select)
+        self.list_box = Gtk.ListBox()
 
-        self.renderer.create_columns(self.treeview)
+        self.list_box.set_selection_mode(Gtk.SelectionMode.NONE)
 
-        self.native = Gtk.ScrolledWindow()
-        self.native.interface = self.interface
-        # same scrolling policies as in the tree widget
-        self.native.set_policy(Gtk.PolicyType.NEVER, Gtk.PolicyType.AUTOMATIC)
-        self.native.add(self.treeview)
-        self.native.set_min_content_width(self.interface.MIN_WIDTH)
-        self.native.set_min_content_height(self.interface.MIN_HEIGHT)
+        self.store = Gio.ListStore()
+        self.list_box.bind_model(self.store, lambda a: a)
 
-    def change_source(self, source):
-        self.treeview.set_model(None)
-        self.store.change_source(source)
-        for i, node in enumerate(source):
-            self.insert(i, node)
+        self.scrolled_window = Gtk.ScrolledWindow()
 
-        self.treeview.set_model(self.store)
+        self.scrolled_window.set_policy(Gtk.PolicyType.NEVER, Gtk.PolicyType.AUTOMATIC)
+        self.scrolled_window.set_min_content_width(self.interface.MIN_WIDTH)
+        self.scrolled_window.set_min_content_height(self.interface.MIN_HEIGHT)
 
-    def insert(self, index, item):
-        self.store.insert(item)
+        self.scrolled_window.add(self.list_box)
 
-    def change(self, item):
-        self.store.change(item)
+        self.refresh_button = RefreshButton(self.scrolled_window.get_vadjustment())
 
-    def remove(self, item, index):
-        self.store.remove(item, index)
+        self.scroll_button = ScrollButton(self.scrolled_window.get_vadjustment())
+        self.scroll_button.set_scroll(lambda: self.scroll_to_row(-1))
+        
+        self.native = Gtk.Overlay()
+        self.native.add_overlay(self.scrolled_window)
 
-    def clear(self):
-        self.store.clear()
+        self.refresh_button.overlay_over(self.native)
+        self.scroll_button.overlay_over(self.native)
 
-    def set_on_refresh(self, handler):
-        pass
+        self.native.interface = self.interface
 
-    def after_on_refresh(self):
-        # No special handling required
-        pass
+        self._on_select_signal_handler = self.list_box.connect(
+            'row-selected', lambda w, item: self.gtk_on_row_selected(item))
+
+        self.right_click_gesture = Gtk.GestureMultiPress.new(self.list_box)
+        self.right_click_gesture.set_button(3)
+        self.right_click_gesture.set_propagation_phase(Gtk.PropagationPhase.BUBBLE)
+        self.right_click_gesture.connect("pressed", self._on_right_click)
```

Comment:
Convention is that any system-specific message handler will have a `gtk_` prefix, so it can be clearly identified.